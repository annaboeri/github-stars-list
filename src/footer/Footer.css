/* Footer Styles */
.footer {
  align-items: center;
  background-color: var(--gray-dark);
  color: white;
  display: flex;
  flex-direction: column;
  justify-content: center;
  padding: 15px 0;
}

.footer p {
  color: white;
  margin: 5px;
}

.footer p > * {
  vertical-align: middle;
}

.footer a, .footer a:visited {
  color: white;
}

.footer a:hover {
  color: var(--green-medium);
}

.logo-container {
  margin: 0 5px;
}

.logo-container:hover .fa-github {
  color: var(--green-medium);
}


.footer-logo {
  height: 25px;
  padding-top: 2px;
}


@media screen and (max-width: 40em) {
  .footer {
    flex-direction: column;
    padding: 20px 10px;
  }
  .logo-conatiner {
    margin-bottom: 0;
    width: 75px;
  }
}
