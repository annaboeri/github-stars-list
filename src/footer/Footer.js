import React from 'react';
import logo from '../github-logo-footer.png';
import './Footer.css';


const Footer = () => {
  return (
    <footer className="footer">
      <p>
        <span>Created by</span>
        <a href="https://github.com/annaboeri" className="logo-container">
          Anna Boeri
        </a>
      </p>
      <p>
        <span>Powered by</span>
        <a href="https://github.com/" className="logo-container">
          <img src={logo} className="footer-logo" alt="github logo" />
        </a>
      </p>
    </footer>
  );
}

export default Footer;
