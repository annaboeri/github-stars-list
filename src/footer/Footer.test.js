import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Footer from './Footer';

describe('Footer Component', () => {
  it('should render correctly', (done) => {

    const component = shallow(<Footer />);
    const tree = toJson(component);

    expect(tree).toMatchSnapshot();
    expect(component.find('.footer').exists()).toBe(true);
    done();
  });
});
