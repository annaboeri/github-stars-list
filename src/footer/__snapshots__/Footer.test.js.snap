// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Component should render correctly 1`] = `
<footer
  className="footer"
>
  <p>
    <span>
      Created by
    </span>
    <a
      className="logo-container"
      href="https://github.com/annaboeri"
    >
      Anna Boeri
    </a>
  </p>
  <p>
    <span>
      Powered by
    </span>
    <a
      className="logo-container"
      href="https://github.com/"
    >
      <img
        alt="github logo"
        className="footer-logo"
        src="github-logo-footer.png"
      />
    </a>
  </p>
</footer>
`;
